src/main/java/fr/catcore/gamegui/mixin/MixinPlayerManager.java

```diff
@@ -31,36 +26,48 @@
-            Collection<CommandNode<ServerCommandSource>> nodes = server.getCommandManager().getDispatcher().getRoot().getChildren();
-            CommandNode<ServerCommandSource> open = null;
-            for (CommandNode<ServerCommandSource> node : nodes) {
-                if (node.getName().equals("game")) {
-                    Collection<CommandNode<ServerCommandSource>> children = node.getChildren();
-                    for (CommandNode<ServerCommandSource> child : children) {
-                        if (child.getName().equals("open")) {
-                            open = node;
-                            break;
-                        }
-                    }
-                }
-            }
-            if (open != null) {
-                MatchableCommand matchableCommand = MatchableCommand.compile(new CommandNode[]{open});
-                if (!(player.inventory.contains(openGame) || CommandPermEvaluator.canUseCommand(serverCommandSource, matchableCommand) != PermissionResult.ALLOW)) {
-                    player.inventory.insertStack(openGame);
-                }
-            }
-        } else {
-            if (!(player.inventory.contains(openGame) || !player.hasPermissionLevel(2))) {
-                player.inventory.insertStack(openGame);
-            }
+        if (!player.inventory.contains(openGame) && this.hasOpenPermission(player)) {
+            player.inventory.insertStack(openGame);
         }
+
         ItemStack joinGame = new ItemStack(GameGuiCustomItems.JOIN_GAME);
         if (!player.inventory.contains(joinGame)) {
             player.inventory.insertStack(joinGame);
         }
     }
+    
+    private boolean hasOpenPermission(ServerPlayerEntity player) {
+        CommandNode<ServerCommandSource> openNode = this.getGameOpenNode();
+
+        if (openNode != null) {
+            ServerCommandSource source = new ServerCommandSource(
+                    player, player.getPos(), player.getRotationClient(),
+                    (ServerWorld) player.world,
+                    player.server.getPermissionLevel(player.getGameProfile()),
+                    player.getName().toString(), player.getDisplayName(),
+                    player.server, player
+            );
+
+            return openNode.getRequirement().test(source);
+        } else {
+            return player.hasPermissionLevel(2);
+        }
+    }
+
+    private CommandNode<ServerCommandSource> getGameOpenNode() {
+        Collection<CommandNode<ServerCommandSource>> nodes = this.server.getCommandManager().getDispatcher().getRoot().getChildren();
+
+        for (CommandNode<ServerCommandSource> node : nodes) {
+            if (node.getName().equals("game")) {
+                Collection<CommandNode<ServerCommandSource>> children = node.getChildren();
+                for (CommandNode<ServerCommandSource> child : children) {
+                    if (child.getName().equals("open")) {
+                        return child;
```

Comment:
side note: found where the perms bug was happening, this was returning `node` before which would mean even people with `/game join` permissions could join :)